### PR TITLE
Am 525 attach document to case

### PR DESCRIPTION
**Pull request by nitishsingla1234**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-525

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/model/CaseDocumentMetadata.java [hunk 1]
```diff
@@ -14,11 +16,11 @@
...
     }
 
-    public CaseDocumentMetadata(String caseId, String caseTypeId, String jurisdictionId, Document document) {
+    public CaseDocumentMetadata(String caseId, String caseTypeId, String jurisdictionId, List<Document> documents) {
         this.caseId = caseId;
         this.caseTypeId = caseTypeId;
         this.jurisdictionId = jurisdictionId;
-        this.document = document;
+        this.documents = documents;
     }
```
Comment (aashishaggarwal123) [inline src/main/java/uk/gov/hmcts/reform/ccd/document/am/model/CaseDocumentMetadata.java]: Please use the new model entity called CaseDocumentsMetadata with document list.

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/service/impl/DocumentManagementServiceImpl.java [hunk 2]
```diff
@@ -182,6 +184,62 @@ public String extractCaseIdFromMetadata(Object storedDocument) {
...
+                                                       .concat(resource.getMetadata().get("caseTypeId")));
+                if (!hashcodeFromStoredDocument.equals(document.getHashToken())) {
+                    throw new ResourceNotFoundException(String.format(": Document %s does not exists in DM Store", document.getId()));
+                }
+            }
+
+            Map<String, String> metadataMap = new HashMap<>();
+            metadataMap.put("jurisdictionId", caseDocumentMetadata.getJurisdictionId());
+            metadataMap.put("caseId", caseDocumentMetadata.getCaseId());
+            metadataMap.put("caseTypeId", caseDocumentMetadata.getCaseTypeId());
```
Comment (aashishaggarwal123) [inline src/main/java/uk/gov/hmcts/reform/ccd/document/am/service/impl/DocumentManagementServiceImpl.java]: We don't need to put entry for the JID and caseTypeId as its already there. this call should only update caseID.

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/service/impl/DocumentManagementServiceImpl.java [hunk 3]
```diff
@@ -320,8 +378,13 @@ public boolean checkUserPermission(ResponseEntity responseEntity, UUID documentI
...
-            return (caseDocumentMetadata.getDocument().getId().equals(documentId.toString())
-                    && caseDocumentMetadata.getDocument().getPermissions().contains(Permission.READ));
+            List<Document> matchingDocuments =
+                caseDocumentMetadata.getDocuments().stream()
+                                    .filter(document -> document.getId().equals(documentId.toString()))
+                                    .filter(document -> document.getPermissions().contains(Permission.READ))
+                                    .collect(Collectors.toList());
+
+            return matchingDocuments.size() > 0;
         }
```
Comment (aashishaggarwal123) [inline src/main/java/uk/gov/hmcts/reform/ccd/document/am/service/impl/DocumentManagementServiceImpl.java]: I think, this block won't be required if we keep the caseDocumentMetadata update with single document.

File: src/main/java/uk/gov/hmcts/reform/ccd/document/am/service/impl/DocumentManagementServiceImpl.java [hunk 4]
```diff
@@ -109,7 +115,7 @@ public ResponseEntity getDocumentMetadata(UUID documentId) {
                 requestEntity,
                 StoredDocumentHalResource.class
                                                                                       );
-            LOG.info("response : " + response.getStatusCode());
+            LOG.error("response : " + response.getStatusCode());
```
[the remark this answers is not in the document]
Comment (nitishsingla1234) [reply src/main/java/uk/gov/hmcts/reform/ccd/document/am/service/impl/DocumentManagementServiceImpl.java]: I will update this.

**Pull request merged**